model/variable.cc

```diff
@@ -480,139 +480,141 @@ bool VariableBase::handleArrows(int dir,bool reset)
 
 void VariableBase::draw(cairo_t *cairo) const
 {
-  double angle=rotation() * M_PI / 180.0;
-  double fm=std::fmod(rotation(),360);
-  float z=zoomFactor();
-
-  RenderVariable rv(*this,cairo);
-  // if rotation is in 1st or 3rd quadrant, rotate as
-  // normal, otherwise flip the text so it reads L->R
-  bool notflipped=(fm>-90 && fm<90) || fm>270 || fm<-270;
-  Rotate r(rotation() + (notflipped? 0: 180),0,0);
-  rv.angle=angle+(notflipped? 0: M_PI);
-
-  // parameters of icon in userspace (unscaled) coordinates
-  float w, h, hoffs, scaleFactor;
-  w=rv.width()*z; 
-  h=rv.height()*z;
-  scaleFactor=max(1.0f,min(0.5f*iWidth()*z/w,0.5f*iHeight()*z/h));
-  if (rv.width()<0.5*iWidth()) w=0.5*iWidth()*z;
-  if (rv.height()<0.5*iHeight()) h=0.5*iHeight()*z;
-  rv.setFontSize(12*scaleFactor*z);
-  hoffs=rv.top()*z;
+  if (!varTabDisplay) {	
```

Comment:
This is not quite right. A variable should only be invisible if varTabDisplay and all outgoing wires are attachedToDefiningVar(). Other items should have their visibility depend on all their outgoing wires too.

It would be reasonable to modify the visible() method for this purpose, as visible() already means whether the item is visible on the canvas.

Reply:
I understand and was confused about whether or not to use visible(). I thought it just hides variables when they are inside groups and displayContents is false.